### ISSUE TITLE
SLP 1.1 has stalled

### ISSUE THREAD

**Issue by mvines** [solana-labs/solana#7904]
Metrics:
* When the tower flatline happened: https://metrics.solana.com:3000/d/testnet/testnet-monitor?orgId=2&from=1579639842339&to=1579639922310&var-testnet=cluster&var-hostid=All
* ~20 minutes later when InvalidEntryHash blocktree errors started to occur:  https://metrics.solana.com:3000/d/testnet/testnet-monitor?orgId=2&from=1579641230438&to=1579641530000&var-testnet=cluster&var-hostid=All&panelId=34

Looks like we had a bad hash (InvalidEntryHash) at or around slot 1033138, which caused a loss of consensus

**Comment by mvines** [comment]
bootstrap leader logs at https://drive.google.com/drive/folders/13246isbjM7DaRX4DpadQd9Be6MwIYXPB?usp=sharing

**Comment by mvines** [comment]
`api-node-syslog.gz`, from the RPC node, at the same location ^

**Comment by mvines** [comment]
bs-leader's last root was 1031214 (leader:       ChorusXqjLC2NbiStKR6k9WoD7wu6TVTtFG8qCL5XBVa)
api node's last root was 1031216 (leader: Fe5sLQAAT7RBT8mcH1AAGCbExJQcYxcwXvp1GjrGbvxs )

**Comment by mvines** [comment]
The cluster forked off at slot 1031271: "The Event".  

Bootstrap leader went one way, these nodes went another way:
```
27SB7d27xvtBJjgsAV8JBDjQroySmZepiNSepeRbRhe9
2tvTYUajoBgeAd66Zhd5Nc2BgKGCgdmasR94fpBokzop
3QZDKya4AHzsLAuRaMRgejrqW6mETnX88aMSkm7FEE7E
3Z5XVczCTXeYeFABoeFm1LngC9657kZMVGNFzqFXviHb
44e8VyWoyZSE2oYHxMHMedAiHkGJqJgPd3tdt6iKoAFL
4MHRFcPheQonBf1pUKrBwJAnn2wP9NEZkXYFEFMfFbWV
4vPqTnfH2ud6hp1yFSFRy9t9xhm8sGDwU4amcZGr2gT7
4ydifDThiWuVtzV92eNGiznuQAnTZtGkb9b2XQoMGAUn
592eDka2qrXWcszC3NNvViKfEyxvuoAbBgohVt75dWq1
5jTcJaq6gLEao1R5rscvfnUhNt6RXg4JFDCegyEhsJG2
7ntcPwcaCSpH66ftVZU5oSuWSpvQfN3kfTDaGUHWsc1m
8wFK4fCAuDoAH1fsgou9yKZPqDMFtJUVoDdkZAAMuhyA
AYZS4CFZRi1165mmUqnpDcHbm1NT9zFGPdjG5VDuK79p
Ah5arzkbkHTMkzUaD5DiCAC1rzxqPgyQDFTnw8Krwz1V
ArpeD4LKYgza1o6aR5xNTQX3hxeik8URxWNQVpA8wirV
B21L2hCrdE4SDhRi2fHKohfSUNAhoLeaWfBp1C9HdF6Y
C8VJytJbZM7KFMXHNUdoF4V7V2QbhkxNs1qYybRoqUEK
CWfPaZJpy8fc2eU7qe1JNnf4oszQFJU68DZiVJGGy4Z7
ChorusXqjLC2NbiStKR6k9WoD7wu6TVTtFG8qCL5XBVa
DaqUBvjHtKYiZ6exUhqrcpDqu5ffYB6QWKwXSwdvDVBj
Daxixc1dFxxLDj85t1CWAsvNXdYq51tDAE51nhPqK9yF
DxLtXrLUrqja3EFjkR4PXNYCuyVtaQnozonCdf3iZk8X
E2bf2uuZ3noWQ9pNSNUdf6Rt6JbPwmrAfCeuhqd98J9S
ETVHRnFkZi7PihPDYibp9fmjfR8P5o7pEs92czku62VV
EduAgutprA7Vp94ZmTU6WRAmqq7VZAXBqH1GyxjWn12D
Ez4iUU87ViJLCnmSy1t1Ti3DLoysFXiBseNfnRfoehyY
FYbyeGqsx8G5mW4p3MfnNEsHaCQQSAmxESf7ct36moGZ
Fe5sLQAAT7RBT8mcH1AAGCbExJQcYxcwXvp1GjrGbvxs
FhacRVSACfKcZNAbvbKuj1MunBKxQu2nHu9raJaGsZzG
G47WACh32JUcxyiCna7UYw45tyYSFKQ58yFpUmhmMybm
GRi3H2M3HxYGAKhz5VrUQipUrAhWj6jTbtjhxiKXHhRj
GsEofbB3rzUK78Ee4NRL6AmcPs6FKRCb7JA8tX6LZjHc
HpzcHxARoR6HtVuZPXWJMLwgusk2UNCan343u6WSQvm2
Luna1VCsPBE4hghuHaL9UFgimBB3V6u6johyd7hGXBL 
SPC3m89qwxGbqYdg1GuaoeZtgJD2hYoob6c4aKLG1zu 
Smith4JYx2otuFgT2dR83qJSfW8RjBZHPsXPyfZBYBu 
pbAxyqHHPMwgEjv8kmjGxysk9rhNtN7q22eAjReq6Hj 
qzCAEHbjb7AtpTPKaetY47LWNLCxFeHuFozjeXhj1k1 
```

**Comment by mvines** [comment]
There's a PDF in the drive link, bs-leader-at-slot-1031600.pdf, that shows the forking after "The Event" from the perspective of the bootstrap leader's ledger.  It was generated by running:
```
$ solana-ledger-tool -l ledger verify --graph-forks bs-leader-at-slot-1031600.pdf --halt-at-slot 1031600
```

**Comment by mvines** [comment]
solana-validator-BKLbrRv7PkopTbWMnZSGS7n4pzXYp5FByFowhBXpX5fN-20200118-155231.log is from brianlong's node, which forked along with the bootstrap leader

**Comment by mvines** [comment]
4vPqTnfH2ud6hp1yFSFRy9t9xhm8sGDwU4amcZGr2gT7.log is from the Certus One node that forked away from the bootstrap leader

**Comment by mvines** [comment]
Fun fact: Luna1VCs-forks.pdf contains the `solana-ledger-tool --graph-forks ... --halt-at-slot 1031600` output for the Luna1VCsPBE4hghuHaL9UFgimBB3V6u6johyd7hGXBL (on the other fork), and it produced the identical output as the bootstrap leader.  So the ledger appears to be consistent between both forks

**Comment by sakridge** [comment]
```
Bad Fork
5KTouxSFVt5d6DyN4oCMASQXzvaeXdwqLNuptRfWAJGH slot_weight: 1031277 17872989769773328688767672   582_536746728294721683326924848 1031276
BKLbrRv7PkopTbWMnZSGS7n4pzXYp5FByFowhBXpX5fN slot_weight: 1031277 17872989769773328688767672  876_8648820087170342578383213580 1031276
9CKPvKr5jZ8rHMQedSFkx474XgSC6Rui1WX5r5kS5ZAk slot_weight: 1031277 17872989769773328688767672 12949551534945470356763487459830 1031276

DxLtXrLUrqja3EFjkR4PXNYCuyVtaQnozonCdf3iZk8X slot_weight: 1031276 17817935403445226877630008  965_617_2900834675956023228946658 1031275

Good Fork
5KTouxSFVt5d6DyN4oCMASQXzvaeXdwqLNuptRfWAJGH slot_weight: 1031280 15206282372332371404410484   582449069257583657751515068256 1031271
BKLbrRv7PkopTbWMnZSGS7n4pzXYp5FByFowhBXpX5fN slot_weight: 1031280 15206282372332371404410484  8768561142616459278646571356988 1031271
9CKPvKr5jZ8rHMQedSFkx474XgSC6Rui1WX5r5kS5ZAk slot_weight: 1031280 15206282372332371404410484 12949463857474759292831675603238 1031271

DxLtXrLUrqja3EFjkR4PXNYCuyVtaQnozonCdf3iZk8X slot_weight: 1031280 15206282372332371404410484  965_610_3096353734665420105857738 1031271

5KTouxSFVt5d6DyN4oCMASQXzvaeXdwqLNuptRfWAJGH slot_weight: 1031291 15206282378332371226011764   582_616338363703313836249988700 1031290
BKLbrRv7PkopTbWMnZSGS7n4pzXYp5FByFowhBXpX5fN slot_weight: 1031291 15206282378332371226011764  876_8728411722578934731306277432 1031290

 DxLtXrLUrqja3EFjkR4PXNYCuyVtaQnozonCdf3iZk8X slot_weight: 1031291 15206282378332371226011764  965_627_0365459854321504840778182 1031290
```

We verified the fork weights are consistent on nodes on the good fork and bad fork. DxLtXr is on the bad fork, rest are on good.

The minor fork has ~52% of the stake voting on it, the 'major' fork has 36% of the stake. The major fork is heaver, but the minor validators do not switch and thus 67% of the network as required by roots are not voting on the fork making progress.

**Comment by sakridge** [comment]
We are able to replay it with the chorus one vote history and ledger. The nodes on the minor fork do not pass the threshold check to be able to vote on the other fork. One reason is they were not able to land many votes on the main trunk and build up their lockup on a common ancestor, so applying the votes to their tower state pops off their most recent votes and the slot it pops off to only has 36% of the stake committed to it.

**Comment by sakridge** [comment]
Another weird thing I see when I run the ledger-tool, there are dropped votes due to hash mismatch. I don't see these in the logs of the nodes though:
```
[2020-01-22T18:32:47.014804739Z INFO  solana_runtime::bank] bank frozen: 1031252 hash: 9G9rZGSvF4Sbh3jpz1GjRNuRyq1KbWHCFxJ8hjM6UUwi
accounts_delta: BankHash 527b393b8bb81be640aa801e02f5048f976f2d6b187
1cad80aaf24e54252cb9e
signature_count: 236
last_blockhash: 5NLApB2QUYeBpbWjezyww2qnwc3wcSoPMGG7mYD9JY12
[2020-01-22T18:32:47.014822743Z INFO  solana_runtime::bank] accounts hash slot: 1031252 stats: BankHashStats { num_removed_accounts: 118, num_added_accounts: 0, num_lamports_stored: 54626177525283, to
tal_data_len: 0, num_executable_accounts: 0 }
```

```
[2020-01-22T18:32:47.039220398Z WARN  solana_vote_program::vote_state] pbAxyqHHPMwgEjv8kmjGxysk9rhNtN7q22eAjReq6Hj dropped vote Vote { slots: [1031252], hash: 67UsWWh9t4TVq8jic9F7QL7ep9kwp1FsarJ7EznKWu7C, timestamp: None }
failed to match hash 67UsWWh9t4TVq8jic9F7QL7ep9kwp1FsarJ7EznKWu7C 9G9rZGSvF4Sbh3jpz1GjRNuRyq1KbWHCFxJ8hjM6UUwi
```

@ryoqun want to take a look?
I uploaded the ledger here: https://drive.google.com/drive/folders/1KcTUsK13yIeJYaoux82iaAmFcZHe9fD1

**Comment by ryoqun** [comment]
@sakridge Odd, I couldn't reproduce the mismatch error using the ledger archive `ledger.tar.gz` in the google drive folder... The voted hash actually matches to the hash of frozen bank on my machine...

In my environment:

```

[2020-01-24T10:24:42.451487645Z INFO  solana_runtime::bank] accounts hash slot: 1031249 stats: BankHashStats { num_removed_accounts: 9, num_added_accounts: 0, num_lamports_stored: 2001003563520, total_data_len: 8280, num_executable_accounts: 0 }
[2020-01-24T10:24:42.455583628Z INFO  solana_runtime::bank] bank frozen: 1031250 hash: 4KQk1WbZVGSyD12Pp9g1465XWZfwr3wvzWp6YQ6SZ9ce accounts_delta: BankHash c7713e4ade5656b61f3eb4920bd889243ee5606cfdf3cbd774641abfe1f0b80f signature_count: 6 last_blockhash: 39qiaTErF8Pfw6nqMpJwdo3YAPxkpj18MZDW1Vke5rA7
[2020-01-24T10:24:42.455624749Z INFO  solana_runtime::bank] accounts hash slot: 1031250 stats: BankHashStats { num_removed_accounts: 6, num_added_accounts: 0, num_lamports_stored: 1001017079840, total_data_len: 6210, num_executable_accounts: 0 }
[2020-01-24T10:24:42.460321512Z INFO  solana_runtime::bank] bank frozen: 1031251 hash: 7HxXJM6kwEyFFBK7afUH4r96L5kzEGkfM7cE75wiVTy6 accounts_delta: BankHash 6910742f254036d744980e520e92d9dfc4150b74a6e2ddf638f9cda5c5b3ebca signature_count: 10 last_blockhash: 3bQs2p5BC4vsc3XyyT9vKZYfKTM3HK3zQtydmASBLs65
[2020-01-24T10:24:42.460364423Z INFO  solana_runtime::bank] accounts hash slot: 1031251 stats: BankHashStats { num_removed_accounts: 8, num_added_accounts: 0, num_lamports_stored: 1100988265440, total_data_len: 6210, num_executable_accounts: 0 }


[2020-01-24T10:24:42.492961951Z INFO  solana_runtime::bank] bank frozen: 1031252 hash: 67UsWWh9t4TVq8jic9F7QL7ep9kwp1FsarJ7EznKWu7C accounts_delta: BankHash 9dc96f0777a487d8b3943e7e07c645c3fe3941401566bff79ef8104474a6c3c8 signature_count: 236 last_blockhash: 5NLApB2QUYeBpbWjezyww2qnwc3wcSoPMGG7mYD9JY12
[2020-01-24T10:24:42.492992652Z INFO  solana_runtime::bank] accounts hash slot: 1031252 stats: BankHashStats { num_removed_accounts: 212, num_added_accounts: 0, num_lamports_stored: 54627615544803, total_data_len: 194580, num_executable_accounts: 0 }


[2020-01-24T10:24:42.501124929Z INFO  solana_runtime::bank] bank frozen: 1031253 hash: 7iMqRTfMZjsbhRoot8J6okGiWqitvVNP9AJM1zdh5YCe accounts_delta: BankHash 1c5c7ef780c1825c5bdfac9b246daa3c896c9434b035e2b434c2b1f35d9a3b7b signature_count: 28 last_blockhash: BUHd43TEteSz3oVL9M3j3z27GmViBGCReCprTXsxch1h
[2020-01-24T10:24:42.501158930Z INFO  solana_runtime::bank] accounts hash slot: 1031253 stats: BankHashStats { num_removed_accounts: 28, num_added_accounts: 0, num_lamports_stored: 6990011581440, total_data_len: 28980, num_executable_accounts: 0 }
[2020-01-24T10:24:42.513456361Z INFO  solana_runtime::bank] bank frozen: 1031254 hash: FkWWDXjhykwcuxgNywSN3ckwn4kN6VLtwozECtKp42Vg accounts_delta: BankHash 479fda9f7ef4fcececec31bf323f80a68c842d1109102b9d1ec3fa52a967e15e signature_count: 74 last_blockhash: EFeDbpxUzFJWsKMPPV3oMVGMUkjqv8TaoNgbZXy8CfN4
[2020-01-24T10:24:42.513491212Z INFO  solana_runtime::bank] accounts hash slot: 1031254 stats: BankHashStats { num_removed_accounts: 71, num_added_accounts: 0, num_lamports_stored: 16629987068320, total_data_len: 70380, num_executable_accounts: 0 }


```

Your quoted log says `total_data_len: 0`. I think this is really odd...
@sakridge Does that still happens in your environment? I want to see surrounding logs as this can't reproduce by me. ;)

**Comment by sakridge** [comment]
@ryoqun OK, interesting. Actually Carl I think was seeing that it's working as well, so maybe it's something about my environment. Let me try and delete the ledger and retry maybe.

An update on the consensus issue, I uploaded a potential fix that we have been considering. It seems to fail in the CI local-cluster although it passes locally for me. Have to look into this issue more to see what is happening. Also, more testing is needed, @carllin is working on a local-cluster test to try and reproduce the original failure which we determined relies on there being two forking states back-to-back.
